src/Featurizers/Components/UnitTests/DocumentStatisticsEstimator_UnitTest.cpp

```diff
@@ -0,0 +1,158 @@
+                                                                            estimator(pAllColumnAnnotations, 0);
+
+    NS::TestHelpers::Train(estimator, inputBatches);
+
+    NS::Featurizers::Components::DocumentStatisticsAnnotationData const &
+                                                                            annotation(estimator.get_annotation_data());
+    return annotation.TotalNumDocuments;
+}
+
+void TestString_Batch() {
+    TestString("this is a document", {"this", "is", "a", "document"});
+    TestString("this   is a   document ", {"this", "is", "a", "document"});
+    TestString(" this is   a document", {"this", "is", "a", "document"});
+    TestString(" this is a document ", {"this", "is", "a", "document"});
+    TestString(" this   is a   document  ", {"this", "is", "a", "document"});
+    TestString("this", {"this"});
+    TestString(" this", {"this"});
+    TestString("this ", {"this"});
+    TestString(" this ", {"this"});
+    TestString("  this  ", {"this"});
+    TestString("", {});
+    TestString(" ", {});
+    TestString("  ", {});
+}
+
+TEST_CASE("string_split") {
+    TestString_Batch();
+}
+
+TEST_CASE("invalid_annotation") {
+    using FrequencyMap = NS::Featurizers::Components::DocumentStatisticsAnnotationData::FrequencyMap;
+    using IndexMap = NS::Featurizers::Components::DocumentStatisticsAnnotationData::IndexMap;
+    FrequencyMap termFreq(FrequencyMap{{"key", 1}});
+    IndexMap termIndex(IndexMap{{"key", 1}});
+    CHECK_THROWS_WITH(NS::Featurizers::Components::DocumentStatisticsAnnotationData(FrequencyMap(),termIndex, 1), "termFreq");
+    CHECK_THROWS_WITH(NS::Featurizers::Components::DocumentStatisticsAnnotationData(termFreq, termIndex, 0), "totalNumDocuments");
+    CHECK_THROWS_WITH(NS::Featurizers::Components::DocumentStatisticsAnnotationData(termFreq, IndexMap(), 1), "termIndex");
+}
+
+TEST_CASE("invalid_trainingpolicy") {
+    using StringDecorator = NS::Featurizers::Components::Details::DocumentStatisticsTrainingOnlyPolicy::StringDecorator;
+    StringDecorator decoratorNull;
+    CHECK_THROWS_WITH(NS::Featurizers::Components::Details::DocumentStatisticsTrainingOnlyPolicy(decoratorNull), "decorator");
+}
+
+TEST_CASE("string_idf") {
+    using FrequencyMap                         = NS::Featurizers::Components::DocumentStatisticsAnnotationData::FrequencyMap;
+
+    FrequencyMap const                         label({{"orange",3}, {"apple", 1}, {"peach", 3}, {"grape", 2}, {"banana",1}});
+    std::vector<std::vector<std::string>> const            inputBatches({{" orange  apple  apple peach  grape "},
+                                                                        {" grape orange     peach peach banana"},
+                                                                        {"orange orange peach   peach orange "}});
+
+    FrequencyMap const                         predict(Test(inputBatches));
+    CHECK(predict == label);
+}
+
+TEST_CASE("string_idf_single_appearance") {
```

Comment:
I'm assuming that each of these tests represents different input scenarios? If so, validate all properties of the annotation data for each of the scenarios.